packages/components/src/components/assets/CoinLogo/CoinLogo.tsx

```diff
@@ -24,15 +25,44 @@ const SvgWrapper = styled.div<Omit<CoinLogoProps, 'symbol'>>`
     }
 `;
 
-export const CoinLogo = ({ symbol, className, size = 32, ...rest }: CoinLogoProps) => (
-    <SvgWrapper className={className} size={size} {...rest}>
-        <ReactSVG
-            src={COINS[symbol]}
-            beforeInjection={svg => {
-                svg.setAttribute('width', `${size}px`);
-                svg.setAttribute('height', `${size}px`);
-            }}
-            loading={() => <span className="loading" />}
-        />
-    </SvgWrapper>
+const StyledIconImage = styled.img<CoinLogoProps>(
+    ({ size }) => `
+        width: ${size}px;
+        height: ${size}px;
+    `,
 );
+
+export const CoinLogo = ({ symbol, iconUrl, className, size = 32, ...rest }: CoinLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    if (symbol) {
+        return (
+            <Wrapper className={className} size={size} {...rest}>
```

Comment:
Not a big deal but `<Wrapper className={className} size={size} {...rest}>` is now duplicated

```ts

return (
    <Wrapper className={className} size={size} {...rest}>
        {symbol ? <ReactSVG .... /> : iconUrl && <> ... </>}
    </Wrapper>
)
```

Would remove the duplication.

Also whole ```ts
 <ReactSVG
                    src={COINS[symbol]}
                    beforeInjection={svg => {
                        svg.setAttribute('width', `${size}px`);
                        svg.setAttribute('height', `${size}px`);
                    }}
                    loading={() => <span className="loading" />}
                />
                ```
Can be extracted as a sub-component with just `symbol` as a property.

Reply:
Missed that, thanks!